pkg/state/state.go

```diff
@@ -311,6 +316,15 @@ BEGIN
 		RETURN;
 	END IF;
 
+	-- Remove any duplicate inferred migrations with the same timestamp for this
+	-- schema. We assume such migrations are multi-statement batched migrations
+	-- and we are only interested in the last one in the batch.
+	DELETE FROM %[1]s.migrations
+	WHERE schema = schemaname
+	AND created_at = current_timestamp
+	AND migration_type = 'inferred'
+	AND migration->'operations'->0->'sql'->>'up' = current_query();
```

Comment:
Would it be safer to avoid inserting it at all?

Reply:
I think we need to insert so that we ensure we insert the final statement in the batch. Otherwise how would we know which was the last one?